MPXPLAY/AU_CARDS/SC_ICH.C

```diff
@@ -177,10 +183,20 @@ static unsigned int snd_intel_buffer_init(struct intel_card_s *card,struct mpxpl
  unsigned int bytes_per_sample=(aui->bits_set>16)? 4:2;
 
  card->pcmout_bufsize=MDma_get_max_pcmoutbufsize(aui,0,ICH_DMABUF_ALIGN,bytes_per_sample,0);
- card->dm=MDma_alloc_cardmem(ICH_DMABUF_MAX_PERIODS*2*sizeof(uint32_t)+card->pcmout_bufsize);
- card->virtualpagetable=(uint32_t *)card->dm->linearptr; // pagetable requires 8 byte align, but dos-allocmem gives 16 byte align (so we don't need alignment correction)
- card->pcmout_buffer=((char *)card->virtualpagetable)+ICH_DMABUF_MAX_PERIODS*2*sizeof(uint32_t);
- aui->card_DMABUFF=card->pcmout_buffer;
+
+ size_t buffer_descriptor_list_size = ICH_DMABUF_MAX_PERIODS*ICH_BDL_ENTRY_SIZE;
+
+ // Allocate Buffer Descriptor List + PCM output buffer in a single allocation
+ card->dm = MDma_alloc_cardmem(buffer_descriptor_list_size + card->pcmout_bufsize);
+
+ // buffer descriptor list requires ICH_BDL_ENTRY_SIZE alignment,
+ // but dos-allocmem gives 16 byte align (so we don't need alignment correction)
+ card->virtualpagetable = (uint32_t *)card->dm->linearptr;
+ card->pcmout_buffer = card->dm->linearptr + buffer_descriptor_list_size;
+
+ // DMA buffer written by MDma_writedata() and MDma_clearbuf()
+ aui->card_DMABUFF = card->pcmout_buffer;
```

Comment:
Just like above, this is unrelated to SIS7012, but documents this part a little better.